### PR TITLE
feat(client): Remove `rejectOnNotFound`

### LINKED ISSUE #19315: Prisma Client: remove `rejectOnNotFound`
We deprecated this option in 4.0.0 and introduced new APIs as the replacement.

The deprecation warning printed at run time currently states that the feature will be removed in Prisma 5:

```tsx
prisma:warn `rejectOnNotFound` option is deprecated and will be removed in Prisma 5. Please use `prisma.model.findUniqueOrThrow` method instead
```

- [ ] Remove code
- [ ] Remove warning
- [ ] Remove tests
- [ ] prisma/docs#4999
- [ ] Rename issue about update and delete (see [slack](https://prisma-company.slack.com/archives/C03MRGP7WLB/p1686155501856429)) https://github.com/prisma/prisma/issues/10142
- [ ] Release notes draft, see https://github.com/prisma/prisma/releases/tag/4.0.0

### PR DESCRIPTION
**Pull request by SevInf**
BREAKING CHANGE: `rejectOnNotFound` option is removed

Close #19315

### CODE REVIEW

File: packages/client/src/runtime/core/model/applyModel.ts [hunk 1]
```diff
@@ -78,8 +77,7 @@ function modelActionsLayer(client: Client, dmmfModelName: string): CompositeProx
     getPropertyValue(key) {
       const dmmfActionName = key as DMMF.ModelAction
 
-      let requestFn = (params: InternalRequestParams) => client._request(params)
-      requestFn = adaptErrors(dmmfActionName, dmmfModelName, client._clientVersion, requestFn)
+      const requestFn = (params: InternalRequestParams) => client._request(params)
```
Comment (millsp) [inline packages/client/src/runtime/core/model/applyModel.ts:80]: Maybe we can remove `requestFn` now to directly use `client._request` directly?

**Review by millsp — APPROVED** [review]

**Review by janpio — CHANGES_REQUESTED** [review]
let's wait with this one

**Review by janpio — COMMENTED** [review]
Removed the block of merge.

File: packages/client/src/runtime/core/model/applyModel.ts (hunk 1 above)
Comment (SevInf) [reply packages/client/src/runtime/core/model/applyModel.ts:80]: We've decided to keep using `NotFoundError` for `*OrThrow` methods for now, so `adaptErrors` will be back soon

**Review by Jolg42 — APPROVED** [review]

**Review by millsp — APPROVED** [review]

**Pull request merged**